### PR TITLE
Fix ATDM Trilinos builds broken by TriBITS update (#10774)

### PR DESCRIPTION
**Pull request by bartlettroscoe**
## Internal issues

* [TRILINOSHD-156](https://sems-atlassian-son.sandia.gov/jira/servicedesk/customer/portal/7/TRILINOSHD-156)
* [TRILINOSHD-157](https://sems-atlassian-son.sandia.gov/jira/servicedesk/customer/portal/7/TRILINOSHD-157)

## Description

This PR contains updates to TriBITS and Trilinos to address failures in the ATDM Trilinos builds described in #10774.  As I fix these builds and update this topic-branch, I will manually merge this topic branch to the 'atdm-nightly-manual-updates' branch so this gets run in the ATDM Trilinos builds.  This way, we can see how this works in the ATDM Trilinos builds without getting help up by issues with PR testing (e.g. #10782).  (See the motivation and description for this workflow in [Restoring productivity through the advanced usage of Git](https://doi.org/10.6084/m9.figshare.12594563).)

Issues addressed in the PR:

* Link errors due to improper order of CGNS and HDF5 libraries on link line described in https://github.com/trilinos/Trilinos/issues/10774#issuecomment-1191702159 (pulls in TriBITSPub/TriBITS#502)
* CUDA build failures due to duplicate CUDA library function definitions when creating libIntercept.so.13.5 described in https://github.com/trilinos/Trilinos/issues/10774#issuecomment-1191995273 (pulls in TriBITSPub/TriBITS#503)
* Fix Link errors missing `__dlopen` for 'ats1' builds (pulls in TriBITSPub/TriBITS#504)

**NOTE:** I also removed Krino from the set of disabled packages since the ATDM customers are not using it and is failing (see #10524).

**NOTE:** This also pulls in TriBITS changes for the PRs:

* https://github.com/TriBITSPub/TriBITS/pull/501
* https://github.com/TriBITSPub/TriBITS/pull/499
* https://github.com/TriBITSPub/TriBITS/pull/498

## Testing

All of the new ATDM Trilinos build errors triggered by triggered by the merge of  #10614 have been resolved as of [2022-07-26](https://testing.sandia.gov/cdash/index.php?project=Trilinos&date=2022-07-26&filtercount=2&showfilters=1&filtercombine=and&field1=groupname&compare1=62&value1=Experimental&field2=buildname&compare2=65&value2=Trilinos-atdm-).  See [below](https://github.com/trilinos/Trilinos/pull/10791#issuecomment-1195407862).

### CODE REVIEW

**Comment by bartlettroscoe** [comment]
FYI: With the merge commit https://github.com/trilinos/Trilinos/commit/bd772e627e28c6df18abd3e0f67bbd1967d66731, we should see the CUDA link errors described [above](https://github.com/trilinos/Trilinos/issues/10774#issuecomment-1191995273) fixed in the ATDM Trilinos builds tomorrow.  (I had to manually merge directly to the 'atdm-nightly' branch because the auto-update completed about 25 minutes ago but none of the ATDM Trilinos builds should have fired off yet so we should still get a consistent set of ATDM Trilinos builds tomorrow.)

**Comment by bartlettroscoe** [comment]
With the merge of this branch to 'atdm-nightly-manual-updates' in commit 36e4968505f0b8e6a74c615ed7c43328fb54a2e9, we should hopefully see all of the new ATDM Trilinos failures caused by the merge of #10614 resolved (and also see the Krino errors go away).

**Comment by bartlettroscoe** [comment]
FYI: With the update to the 'atdm-nightly' branch yesterday, the all of the 'ats1' configurations on 'mutrino' now all build successfully as shown for the ATDM Trilinos builds [today 2022-07-26](https://testing.sandia.gov/cdash/index.php?project=Trilinos&date=2022-07-26&filtercount=3&showfilters=1&filtercombine=and&field1=groupname&compare1=62&value1=Experimental&field2=buildname&compare2=65&value2=Trilinos-atdm-&field3=buildname&compare3=63&value3=ats1).  There are 6 new test failures for for the build [`Trilinos-atdm-ats1-hsw_intel-19.0.4_mpich-7.7.15_openmp_static_opt`](https://testing.sandia.gov/cdash/index.php?project=Trilinos&parentid=11169518&filtercount=1&showfilters=1&field1=groupname&compare1=62&value1=Experimental&filtercombine=and) but they all show srun creation failures:

```
srun: Job 27675822 step creation temporarily disabled, retrying (Requested nodes are busy)
srun: Job 27675822 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27675822 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27675822 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27675822 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27675822 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27675822 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27675822 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27675822 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27675822 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27675822 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27675822 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27675822 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27675822 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27675822 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27675822 step creation still disabled, retrying (Requested nodes are busy)
srun: Job 27675822 step creation still disabled, retrying (Requested nodes are busy)
```

This looks like failures we have seen before on 'mutrino' as reported in #3942 and #3499.  Not sure if this is related to the TriBITS updates or some other changes or if this is just a fluke.

As for the 4 tests failing in the build [`Trilinos-atdm-ats1-knl_intel-19.0.4_mpich-7.7.15_openmp_static_opt`](https://testing.sandia.gov/cdash/index.php?project=Trilinos&parentid=11169454&filtercount=1&showfilters=1&field1=groupname&compare1=62&value1=Experimental&filtercombine=and), those tests were failing before the TriBITS upgrade as shown in [this query](https://testing.sandia.gov/cdash/index.php?project=Trilinos&begin=2022-06-01&end=2022-07-26&filtercount=2&showfilters=1&filtercombine=and&field1=groupname&compare1=62&value1=Experimental&field2=buildname&compare2=61&value2=Trilinos-atdm-ats1-knl_intel-19.0.4_mpich-7.7.15_openmp_static_opt) and comparing the test failures on testing day [2022-07-18](https://testing.sandia.gov/cdash/viewTest.php?onlyfailed&buildid=11155794) and those on  
testing day [2022-07-26](https://testing.sandia.gov/cdash/viewTest.php?onlyfailed&buildid=11169454).

**Comment by bartlettroscoe** [comment]
All of the new ATDM Trilinos build errors due to the TriBITS upgrade from #10614 now appear to be resolved as of today, ATDM Trilinos testing day [2022-07-26](https://testing.sandia.gov/cdash/index.php?project=Trilinos&date=2022-07-26&filtercount=2&showfilters=1&filtercombine=and&field1=groupname&compare1=62&value1=Experimental&field2=buildname&compare2=65&value2=Trilinos-atdm-).  The only remaining failures are those that already existed which include configure errors for three SPARC 'mini' builds and build errors for the 'tlcc2' builds as reported at the top of #10774.

So this PR is ready to merge and it fixes all of the ATDM Trilinos builds.

**Comment by bartlettroscoe** [comment]
Here are the notes for my local builds reproducing and fixing the CGNS/HDF5 link errors I had [above](https://github.com/trilinos/Trilinos/pull/10791#issue-1313857639).

----

I was able to reproduce the CGNS/HDF5 link errors described inhttps://github.com/trilinos/Trilinos/issues/10774#issuecomment-1191702159 and then verify the fix on the machine 'ceerws1113' with the build `cee-rhel7_intel-19.0.3_intelmpi-2018.4_serial_static_opt`.  Details are in expanded section below.

<details>

<summary><b>'cee-rhel7_intel-19.0.3_intelmpi-2018.4_serial_static_opt' build on  'ceerws1113'</b> (click to expand)</summary>

<br>

For an earlier version of this branch (before putting this into TriBITS proper and snapshotting TriBITS 'master' back in):

```
$ git log-short --name-status HEAD --not github/develop
07bae560f32 "WIP: Correctly handle abs dir for findmod for TPL dependencies file (#10774)"
Author: Roscoe A. Bartlett <rabartl@sandia.gov>
Date:   Thu Jul 21 11:00:55 2022 -0600 (5 hours ago)

M       cmake/tribits/core/package_arch/TribitsReadDepsFilesCreateDepsGraph.cmake

d740429a585 "Add dependency of CGNS on HDF5 (#10774)"
Author: Roscoe A. Bartlett <rabartl@sandia.gov>
Date:   Thu Jul 21 10:35:22 2022 -0600 (5 hours ago)

A       cmake/tribits/common_tpls/FindTPLCGNSDependencies.cmake

39965623e25 "Disable Krino in ATDM Trilinos builds (#10524)"
Author: Roscoe A. Bartlett <rabartl@sandia.gov>
Date:   Thu Jul 21 10:14:14 2022 -0600 (6 hours ago)

M       cmake/std/atdm/ATDMDisables.cmake
```

On 'ceerws1113' I ran:

```
$ ssh ceerws1113

$ cd /fgs/rabartl/Trilinos.base/BUILDS/ATDM/CHECKIN/

$ env \
    ATDM_CONFIG_BUILD_COUNT_OVERRIDE=24 \
    ATDM_CONFIG_CTEST_PARALLEL_LEVEL_OVERIDE=16 \
  ./checkin-test-atdm-cee-rhel7.sh \
    cee-rhel7_intel-19.0.3_intelmpi-2018.4_serial_static_opt \
    --enable-all-packages=on --local-do-all

$ grep -A 100 "failed out of" cee-rhel7_intel-19.0.3_intelmpi-2018.4_serial_static_opt/ctest.out 
100% tests passed, 0 tests failed out of 2511

Subproject Time Summary:
Adelus                     =   4.58 sec*proc (4 tests)
Amesos2                    =  13.64 sec*proc (8 tests)
Belos                      = 153.18 sec*proc (144 tests)
Ifpack2                    =  47.41 sec*proc (53 tests)
Intrepid2                  = 322.66 sec*proc (183 tests)
Kokkos                     =  97.46 sec*proc (39 tests)
KokkosKernels              =  96.73 sec*proc (17 tests)
Krino                      =   5.33 sec*proc (1 test)
MueLu                      = 298.76 sec*proc (113 tests)
NOX                        = 154.86 sec*proc (110 tests)
Panzer                     = 2402.23 sec*proc (218 tests)
Phalanx                    =  12.06 sec*proc (36 tests)
ROL                        = 753.13 sec*proc (207 tests)
Rythmos                    =  48.12 sec*proc (83 tests)
SEACAS                     = 129.70 sec*proc (66 tests)
STK                        =   4.34 sec*proc (1 test)
Sacado                     =  42.73 sec*proc (299 tests)
ShyLU_Node                 =   4.59 sec*proc (1 test)
Stratimikos                =  85.10 sec*proc (41 tests)
Teko                       = 145.51 sec*proc (18 tests)
Tempus                     = 293.13 sec*proc (157 tests)
Teuchos                    =  60.93 sec*proc (140 tests)
Thyra                      =  27.32 sec*proc (82 tests)
Tpetra                     = 183.74 sec*proc (271 tests)
TrilinosATDMConfigTests    =  91.91 sec*proc (12 tests)
TrilinosBuildStats         =   4.47 sec*proc (2 tests)
Xpetra                     =  10.11 sec*proc (19 tests)
Zoltan2                    = 156.05 sec*proc (186 tests)

Total Test time (real) = 382.54 sec
```

Note that the build failed the first time due to a compiler crash and I finished running the build and tests with:

```
$ cd cee-rhel7_intel-19.0.3_intelmpi-2018.4_serial_static_opt/

$ time make NP=24 &> make.out
real    19m2.040s
user    372m48.994s
sys     34m21.185s

$ time ctest -j16 &> ctest.out 

real    6m22.668s
user    154m58.596s
sys     16m38.432s
```

</details>

I tested the other changes in this PR in:

* https://github.com/trilinos/Trilinos/issues/10774#issuecomment-1192805563
* https://github.com/trilinos/Trilinos/issues/10774#issuecomment-1194856254

**Comment by bartlettroscoe** [comment]
@rppawlo and @keitat,

Can one of you please approve this PR to allow it to merge to the Trilinos 'develop' branch?  I have verified it fixes all of the ATDM Trilinos build errors caused by the initial merge of #10614.  (That was verified by manually merging this topic branch to the 'atdm-nightly' branch and watching the ATDM Trilinos builds as described [above](https://github.com/trilinos/Trilinos/pull/10791#issuecomment-1195407862)).

This fixes the SPARC Trilinos Integration builds and merging this will allow SPARC to upgrade to a new version of Trilinos 'develop'.

This will also likely fix many other customer builds of Trilinos on various platforms, especially CUDA builds on some platforms.

**Review by rppawlo — APPROVED** [review]

**Review by keitat — APPROVED** [review]

**Comment by bartlettroscoe** [comment]
FYI: There are no results shown on CDash [here](https://trilinos-cdash.sandia.gov/index.php?project=Trilinos&filtercombine=and&filtercombine=and&filtercombine=and&filtercombine=and&filtercombine=and&filtercombine=and&filtercount=2&showfilters=1&filtercombine=and&field1=buildname&compare1=63&value1=PR-10791-test&field2=buildstarttime&compare2=84&value2=NOW).

**Comment by bartlettroscoe** [comment]
FYI: The last two PR testing iteration shored STARTING and FAILED in less than 10 minutes.  So everything is just crashing and not even getting to the build or even the configures likely.

**Comment by PhilMiller** [comment]
This seems to fix an issue EMPIRE was seeing of a link failure in percept against a missing cublas:

```
cmake -D Trilinos_CONFIGURE_OPTIONS_FILE:STRING=cmake/std/atdm/ATDMDevEnv.cmake,cmake/std/atdm/apps/empire/EMPIRETrilinosEnables.cmake -D Panzer_ENABLE_EXAMPLES:BOOL=ON -D Panzer_ENABLE_TESTS:BOOL=ON -D TPL_ENABLE_gtest=OFF -D Trilinos_ENABLE_Tempus=OFF -D Sacado_ENABLE_HIERARCHICAL_DFAD:BOOL=ON  -D TPL_ENABLE_CUSPARSE:BOOL=OFF -G Ninja ~/repos/Trilinos/
```

```
21:18:29  [2971/4115] Linking CXX executable packages/percept/perceptX
21:18:29  FAILED: packages/percept/perceptX 
21:18:29  : && /projects/sems/install/rhel7-x86_64/sems/compiler/gcc/7.2.0/openmpi/1.10.1/bin/mpicxx --remove-duplicate-link-files  --relocatable-device-code=true  -expt-extended-lambda -arch=sm_70  -O3 -DNDEBUG  packages/percept/CMakeFiles/perceptX.dir/src/adapt/main/AdaptMain.cpp.o packages/percept/CMakeFiles/perceptX.dir/src/adapt/main/MeshAdapt.cpp.o -o packages/percept/perceptX  -Wl,-rpath,::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::  packages/percept/libpercept.a  packages/stk/stk_io/stk_io/util/libstk_io_util.a  packages/stk/stk_io/stk_io/libstk_io.a  packages/seacas/libraries/ioss/src/main/libio_info_lib.a  packages/seacas/libraries/ioss/src/init/libIonit.a  packages/seacas/libraries/ioss/src/transform/libIotr.a  packages/seacas/libraries/ioss/src/heartbeat/libIohb.a  packages/seacas/libraries/ioss/src/gen_struc/libIogs.a  packages/seacas/libraries/ioss/src/text_mesh/libIotm.a  packages/seacas/libraries/ioss/src/generated/libIogn.a  packages/seacas/libraries/ioss/src/visualization/libIovs.a  packages/seacas/libraries/ioss/src/cgns/libIocgns.a  packages/seacas/libraries/ioss/src/pamgen/libIopg.a  packages/seacas/libraries/ioss/src/exodus/libIoex.a  packages/seacas/libraries/ioss/src/libIoss.a  packages/pamgen/src/libpamgen_extras.a  packages/pamgen/src/libpamgen.a  packages/zoltan/src/libzoltan.a  -lm  /projects/sems/install/rhel7-x86_64/sems/tpl/cgns/4.1.1/gcc/7.2.0/openmpi/1.10.1/parallel/lib/libcgns.a  packages/stk/stk_mesh/stk_mesh/base/libstk_mesh_base.a  packages/stk/stk_topology/stk_topology/libstk_topology.a  packages/stk/stk_expreval/stk_expreval/libstk_expreval.a  packages/stk/stk_transfer/stk_transfer/libstk_transfer_impl.a  packages/stk/stk_search/stk_search/libstk_search.a  packages/stk/stk_math/stk_math/libstk_math.a  packages/stk/stk_util/stk_util/command_line/libstk_util_command_line.a  packages/stk/stk_util/stk_util/diag/libstk_util_diag.a  packages/stk/stk_util/stk_util/registry/libstk_util_registry.a  packages/stk/stk_util/stk_util/environment/libstk_util_env.a  packages/stk/stk_util/stk_util/parallel/libstk_util_parallel.a  packages/stk/stk_util/stk_util/util/libstk_util_util.a  packages/seacas/libraries/aprepro_lib/libaprepro_lib.a  packages/seacas/libraries/exodus/libexodus.a  /projects/sems/install/rhel7-x86_64/sems/tpl/hdf5/1.10.6/gcc/7.2.0/openmpi/1.10.1/parallel/lib/libhdf5_hl.a  /projects/sems/install/rhel7-x86_64/sems/tpl/hdf5/1.10.6/gcc/7.2.0/openmpi/1.10.1/parallel/lib/libhdf5.a  /projects/sems/install/rhel7-x86_64/sems/tpl/zlib/1.2.8/gcc/7.2.0/base/lib/libz.a  -ldl  /projects/sems/install/rhel7-x86_64/sems/tpl/netcdf/4.7.3/gcc/7.2.0/openmpi/1.10.1/parallel/lib/libnetcdf.a  /projects/sems/install/rhel7-x86_64/sems/tpl/netcdf/4.7.3/gcc/7.2.0/openmpi/1.10.1/parallel/lib/libpnetcdf.a  /projects/sems/install/rhel7-x86_64/sems/tpl/hdf5/1.10.6/gcc/7.2.0/openmpi/1.10.1/parallel/lib/libhdf5_hl.a  /projects/sems/install/rhel7-x86_64/sems/tpl/hdf5/1.10.6/gcc/7.2.0/openmpi/1.10.1/parallel/lib/libhdf5.a  /projects/sems/install/rhel7-x86_64/sems/tpl/zlib/1.2.8/gcc/7.2.0/base/lib/libz.a  -ldl  -lcurl  packages/kokkos-kernels/src/libkokkoskernels.a  packages/kokkos/algorithms/src/libkokkosalgorithms.a  packages/kokkos/containers/src/libkokkoscontainers.a  packages/intrepid/src/libintrepid.a  packages/shards/src/libshards.a  packages/teuchos/remainder/src/libteuchosremainder.a  packages/sacado/src/libsacado.a  packages/teuchos/numerics/src/libteuchosnumerics.a  /usr/lib64/liblapack.so.3  /usr/lib64/libblas.so.3  packages/teuchos/kokkoscomm/src/libteuchoskokkoscomm.a  packages/teuchos/comm/src/libteuchoscomm.a  packages/teuchos/kokkoscompat/src/libteuchoskokkoscompat.a  packages/teuchos/parameterlist/src/libteuchosparameterlist.a  packages/teuchos/parser/src/libteuchosparser.a  packages/teuchos/core/src/libteuchoscore.a  packages/kokkos/core/src/libkokkoscore.a  -ldl && :
21:18:29  Exists: /projects/sems/install/rhel7-x86_64/sems/tpl/zlib/1.2.8/gcc/7.2.0/base/lib/libz.a
21:18:29  Exists: /projects/sems/install/rhel7-x86_64/sems/tpl/hdf5/1.10.6/gcc/7.2.0/openmpi/1.10.1/parallel/lib/libhdf5.a
21:18:29  Exists: /projects/sems/install/rhel7-x86_64/sems/tpl/hdf5/1.10.6/gcc/7.2.0/openmpi/1.10.1/parallel/lib/libhdf5_hl.a
21:18:29  packages/kokkos-kernels/src/libkokkoskernels.a(Sparse_sptrsv_solve_eti_DOUBLE_ORDINAL_INT_OFFSET_INT_LAYOUTLEFT_EXECSPACE_CUDA_MEMSPACE_CUDASPACE.cpp.o): In function `void KokkosBlas::trsm<Kokkos::View<double**, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::Cuda, Kokkos::CudaUVMSpace>, Kokkos::MemoryTraits<1u> >, Kokkos::View<double**, Kokkos::LayoutLeft, Kokkos::CudaUVMSpace, Kokkos::MemoryTraits<1u> > >(char const*, char const*, char const*, char const*, Kokkos::View<double**, Kokkos::LayoutLeft, Kokkos::CudaUVMSpace, Kokkos::MemoryTraits<1u> >::const_value_type&, Kokkos::View<double**, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::Cuda, Kokkos::CudaUVMSpace>, Kokkos::MemoryTraits<1u> > const&, Kokkos::View<double**, Kokkos::LayoutLeft, Kokkos::CudaUVMSpace, Kokkos::MemoryTraits<1u> > const&)':
21:18:29  tmpxft_00022b13_00000000-5_Sparse_sptrsv_solve_eti_DOUBLE_ORDINAL_INT_OFFSET_INT_LAYOUTLEFT_EXECSPACE_CUDA_MEMSPACE_CUDASPACE.cudafe1.cpp:(.text._ZN10KokkosBlas4trsmIN6Kokkos4ViewIPPdJNS1_10LayoutLeftENS1_6DeviceINS1_4CudaENS1_12CudaUVMSpaceEEENS1_12MemoryTraitsILj1EEEEEENS2_IS4_JS5_S8_SB_EEEEEvPKcSF_SF_SF_RNT0_16const_value_typeERKT_RKSG_[_ZN10KokkosBlas4trsmIN6Kokkos4ViewIPPdJNS1_10LayoutLeftENS1_6DeviceINS1_4CudaENS1_12CudaUVMSpaceEEENS1_12MemoryTraitsILj1EEEEEENS2_IS4_JS5_S8_SB_EEEEEvPKcSF_SF_SF_RNT0_16const_value_typeERKT_RKSG_]+0x2c5): undefined reference to `cublasDtrsm_v2'
21:18:29  packages/kokkos-kernels/src/libkokkoskernels.a(Sparse_sptrsv_solve_eti_DOUBLE_ORDINAL_INT_OFFSET_INT_LAYOUTLEFT_EXECSPACE_CUDA_MEMSPACE_CUDASPACE.cpp.o): In function `KokkosBlas::Impl::Scal<Kokkos::View<double*, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::Cuda, Kokkos::CudaSpace>, Kokkos::MemoryTraits<1u> >, double, Kokkos::View<double const*, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::Cuda, Kokkos::CudaSpace>, Kokkos::MemoryTraits<1u> >, 1, true, true>::scal(Kokkos::View<double*, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::Cuda, Kokkos::CudaSpace>, Kokkos::MemoryTraits<1u> > const&, double const&, Kokkos::View<double const*, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::Cuda, Kokkos::CudaSpace>, Kokkos::MemoryTraits<1u> > const&)':
21:18:29  tmpxft_00022b13_00000000-5_Sparse_sptrsv_solve_eti_DOUBLE_ORDINAL_INT_OFFSET_INT_LAYOUTLEFT_EXECSPACE_CUDA_MEMSPACE_CUDASPACE.cudafe1.cpp:(.text._ZN10KokkosBlas4Impl4ScalIN6Kokkos4ViewIPdJNS2_10LayoutLeftENS2_6DeviceINS2_4CudaENS2_9CudaSpaceEEENS2_12MemoryTraitsILj1EEEEEEdNS3_IPKdJS5_S9_SB_EEELi1ELb1ELb1EE4scalERKSC_RSD_RKSF_[_ZN10KokkosBlas4Impl4ScalIN6Kokkos4ViewIPdJNS2_10LayoutLeftENS2_6DeviceINS2_4CudaENS2_9CudaSpaceEEENS2_12MemoryTraitsILj1EEEEEEdNS3_IPKdJS5_S9_SB_EEELi1ELb1ELb1EE4scalERKSC_RSD_RKSF_]+0x159): undefined reference to `cublasDscal_v2'
21:18:29  packages/kokkos-kernels/src/libkokkoskernels.a(Sparse_sptrsv_solve_eti_DOUBLE_ORDINAL_INT_OFFSET_INT_LAYOUTLEFT_EXECSPACE_CUDA_MEMSPACE_CUDAUVMSPACE.cpp.o): In function `KokkosBlas::Impl::Scal<Kokkos::View<double*, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::Cuda, Kokkos::CudaUVMSpace>, Kokkos::MemoryTraits<1u> >, double, Kokkos::View<double const*, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::Cuda, Kokkos::CudaUVMSpace>, Kokkos::MemoryTraits<1u> >, 1, true, true>::scal(Kokkos::View<double*, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::Cuda, Kokkos::CudaUVMSpace>, Kokkos::MemoryTraits<1u> > const&, double const&, Kokkos::View<double const*, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::Cuda, Kokkos::CudaUVMSpace>, Kokkos::MemoryTraits<1u> > const&)':
21:18:29  tmpxft_00022b4e_00000000-5_Sparse_sptrsv_solve_eti_DOUBLE_ORDINAL_INT_OFFSET_INT_LAYOUTLEFT_EXECSPACE_CUDA_MEMSPACE_CUDAUVMSPACE.cudafe1.cpp:(.text._ZN10KokkosBlas4Impl4ScalIN6Kokkos4ViewIPdJNS2_10LayoutLeftENS2_6DeviceINS2_4CudaENS2_12CudaUVMSpaceEEENS2_12MemoryTraitsILj1EEEEEEdNS3_IPKdJS5_S9_SB_EEELi1ELb1ELb1EE4scalERKSC_RSD_RKSF_[_ZN10KokkosBlas4Impl4ScalIN6Kokkos4ViewIPdJNS2_10LayoutLeftENS2_6DeviceINS2_4CudaENS2_12CudaUVMSpaceEEENS2_12MemoryTraitsILj1EEEEEEdNS3_IPKdJS5_S9_SB_EEELi1ELb1ELb1EE4scalERKSC_RSD_RKSF_]+0x159): undefined reference to `cublasDscal_v2'
21:18:29  packages/kokkos-kernels/src/libkokkoskernels.a(KokkosBlas_Cuda_tpl.cpp.o): In function `KokkosBlas::Impl::CudaBlasSingleton::CudaBlasSingleton()':
21:18:29  tmpxft_00024347_00000000-5_KokkosBlas_Cuda_tpl.cudafe1.cpp:(.text+0x11a): undefined reference to `cublasCreate_v2'
21:18:29  packages/kokkos-kernels/src/libkokkoskernels.a(KokkosBlas_Cuda_tpl.cpp.o): In function `std::_Function_handler<void (), KokkosBlas::Impl::CudaBlasSingleton::CudaBlasSingleton()::{lambda()#1}>::_M_invoke(std::_Any_data const&)':
21:18:29  tmpxft_00024347_00000000-5_KokkosBlas_Cuda_tpl.cudafe1.cpp:(.text+0x47): undefined reference to `cublasDestroy_v2'
21:18:29  collect2: error: ld returned 1 exit status
```

**Comment by bartlettroscoe** [comment]
> This seems to fix an issue EMPIRE was seeing of a link failure in percept against a missing cublas:

@PhilMiller, good to hear.  We have been delayed getting this merged due to issues with PR testing.

Does EMPIRE still have a workflow that would allow directly merging this branch into your EMPIRE Trilinos fork repo branch?  If so, that is an easy solution.

**Comment by bartlettroscoe** [comment]
NOTE: I put `AT: RETEST` on this PR 11 hours ago and no testing has started yet.  The reason for this is likely given in https://github.com/trilinos/Trilinos/issues/3276#issuecomment-1204404357.

**Comment by PhilMiller** [comment]
> > This seems to fix an issue EMPIRE was seeing of a link failure in percept against a missing cublas:
> 
> @PhilMiller, good to hear. We have been delayed getting this merged due to issues with PR testing.
> 
> Does EMPIRE still have a workflow that would allow directly merging this branch into your EMPIRE Trilinos fork repo branch? If so, that is an easy solution.

It'll take a bit of manual effort, but that's likely to happen if this doesn't make it through to Trilinos `develop` in the next day or so

**Comment by bartlettroscoe** [comment]
FYI: I pushed commit https://github.com/trilinos/Trilinos/pull/10791/commits/ef3b1957e41f390acf28deba06d66b7227423aea to see if this changes the results as per #10840.

**Comment by bartlettroscoe** [comment]
FYI: I created the issue [TRILINOSHD-156](https://sems-atlassian-son.sandia.gov/jira/servicedesk/customer/portal/7/TRILINOSHD-156) requesting:

----

Hello Trilinos Framework team,

Can someone please manually merge the PR:

https://github.com/trilinos/Trilinos/pull/10791

It fixes issues for several customers. It is approved to be merged and it passed both of its PR testing iterations except for a randomly failing Tpetra that is impacting several PRs (and therefore the failure has nothing to do with the changes in PR #10791, see: https://github.com/trilinos/Trilinos/issues/10847).

With the logjam going on with PR testing over the last several weeks and given that it takes upwards of 3 days to more to get a PR build to run, one can argue that merging PRs that are clearly okay to merge and remove them from PR testing is a good thing to do.

Thanks,

-Ross

**Comment by bartlettroscoe** [comment]
NOTE: The branch [10774-install-run-demo](https://github.com/bartlettroscoe/Trilinos/tree/10774-install-run-demo) for PR #10813 was created from this PR branch [bartlettroscoe:10774-fix-atdm-builds](https://github.com/bartlettroscoe/Trilinos/tree/10774-fix-atdm-builds).  All of the unique commits on this PR shown [here](https://github.com/trilinos/Trilinos/pull/10791/commits) with the top unique commit https://github.com/trilinos/Trilinos/pull/10791/commits/042bdd7648a69d816ed42a4e76d983be498fd8b4 as shown as the initial commits in the branch [bartlettroscoe:10774-install-run-demo](https://github.com/bartlettroscoe/Trilinos/tree/10774-install-run-demo) for PR #10813 shown [here](https://github.com/trilinos/Trilinos/pull/10813/commits) (again, the same commit https://github.com/trilinos/Trilinos/pull/10813/commits/042bdd7648a69d816ed42a4e76d983be498fd8b4 is shown in the PR #10813).

Therefore, when PR #10813 was merged, it effectively merged this PR branch.

SIDE NOTE: I have no idea why GitHub is showing a conflict with the file `make/tribits/core/package_arch/TribitsExternalPackageWriteConfigFile.cmake`:

image

When I merge this topic branch with 'develop' locally, it merges fine as shown by:

```
$ cd Trilinos/

$ git checkout 10774-fix-atdm-builds

$ git log-short --name-status -1
ef3b1957e41 "Merge remote-tracking branch 'github/develop' into 10774-fix-atdm-builds (#10791, #10840)"
Author: Roscoe A. Bartlett <rabartl@sandia.gov>
Date:   Mon Aug 8 12:23:19 2022 -0600 (4 days ago)

$ git fetch github

$ git log-short --name-status -1 github/develop
e061ffc182a "Merge Pull Request #10813 from bartlettroscoe/Trilinos/10774-install-run-demo"
Author: trilinos-autotester <trilinos@sandia.gov>
Date:   Fri Aug 12 08:52:45 2022 -0600 (4 hours ago)

$ git merge --no-edit github/develop

$ git log -1
commit ea6c74261caaac6f5f727a8f012007208f5e1c2c (HEAD -> 10774-fix-atdm-builds)
Merge: ef3b1957e41 e061ffc182a
Author: Roscoe A. Bartlett <rabartl@sandia.gov>
Date:   Fri Aug 12 12:32:15 2022 -0600

    Merge remote-tracking branch 'github/develop' into 10774-fix-atdm-builds
```

So, in other words, GitHub seems to have a defect so we can ignore this.

Therefore, because all of the unique commits from this PR were merged along with PR #10813, we can close this PR and mark it as merged.
